### ISSUE TITLE
Consider adding support for pre-commit

### ISSUE THREAD

**Issue by swick** [bilelmoussaoui/gobject-lint#75]
I'm using https://pre-commit.com/ in xdg-desktop-portal to run a bunch of linters on every commit, to catch mistakes as early as possible. It would be nice if there was a goblint hook for it so I don't have to write a bunch of shell scripts.

**Comment by bilelmoussaoui** [comment]
after having a look at pre-commit hooks, it seems that it would install goblint from main with cargo, is that desired?

**Comment by swick** [comment]
Yes, I think that should be fine. It's possible to define the commit to check out (rev), so that's how I would keep it fixed.

It would be even nicer if it used the image. Something like

```yaml
- id: goblint
  name: "goblint"
  entry: podman run ...
  language: system
```

could work, but you'd have to update the image tag then all the time which is a bit meh.

**Comment by swick** [comment]
Actually as a user I can use the local repo with `language: docker_image` (https://github.com/pre-commit/pre-commit/blob/main/pre_commit/languages/docker_image.py) to just run the gotblint image. It's currently broken for podman but that seems fixable.

Maybe nothing is needed here after all.

**Comment by swick** [comment]
Probably enough to document that this works for pre-commit:

```yaml
- repo: local
  hooks:
  - id: goblint
    name: goblint linter
    language: docker_image
    entry: ghcr.io/bilelmoussaoui/goblint:main .
    verbose: true
    pass_filenames: false
```

With one caveat: it is currently broken when using podman (https://github.com/pre-commit/pre-commit/pull/3676).

**Comment by bilelmoussaoui** [comment]
seems like it will stay broken :P Mind sending a patch to document this on readme?

**Comment by swick** [comment]
It's literally not working on podman no matter what you do. The only workaround is not using `language: docker` but execute podman directly via `language: unsupported` but that obviously won't work on docker. Not every day you learn that the sole maintainer of a hugely popular project is an illiterate buffoon.

**Comment by swick** [comment]
This is what I'm using now,  but it might be a bit too specific to put into the docs.

```yaml
- repo: local
  hooks:
  - id: goblint
    name: goblint linter
    language: unsupported
    entry: .githooks/run-podman.sh ghcr.io/bilelmoussaoui/goblint:main-f65b232 .
    files: \.[ch]$
    pass_filenames: false
```

```sh
#!/usr/bin/env bash

set -eo pipefail
set -x

RUN="${TOOLBOX_PATH:+flatpak-spawn --host}"

if $RUN command -v podman &>/dev/null; then
  exec $RUN podman run --rm --workdir "/src" -v "$PWD:/src:Z" $@
else
  echo '`podman` not found.'
  exit 1
fi
```

**Comment by bilelmoussaoui** [comment]
I think we can  close this one and point people to it if someone opens a similar issue in the future.
